bootstrap/bootstrap.go

```diff
@@ -405,6 +404,12 @@ func (b *Bootstrap) applyEnvironmentChanges(environ *env.Environment, dir string
 		return
 	}
 
+	mergedEnv := b.shell.Env.Merge(environ)
+
+	// reset output redactors based on new environment variable values
+	redactors.Flush()
```

Comment:
It's necessary to `Flush()` the redactor before resetting it, otherwise we'll lose/confuse any partial output that hadn't been terminated with a newline etc. It would be nice if the redactor implementation dealt with this detail, but I wasn't willing to go that deep right now.